purchase_packaging/models/procurement_order.py

```diff
@@ -18,7 +18,7 @@ def _prepare_purchase_order_line(self, po, supplier):
         seller = self.product_id._select_seller(
             partner_id=supplier.name,
             quantity=res['product_qty'],
-            date=po.date_order and fields.Date.from_string(po.date_order),
+            date=po.date_order or None,
```

Comment:
@tfossoulw Into the server the date_order is always a string! Therefore the fix must be:
```python
date=po.date_order and fields.Date.from_string(po.date_order) or None,
```

Reply:
On the _select_seller, it compare "seller.date_start" with "date".
Ex: if seller.date_start and seller.date_start > date:

In this way  "seller.date_start" is a String and is not converted in to a date. So if we send a date format it will not work.


I tested it and I got this error : 
  File "/home/odoo/wink/colombo_odoo/odoo-colombo/src/odoo/odoo/addons/product/models/product.py", line 508, in _select_seller
    if seller.date_start and seller.date_start > date:
TypeError: can't compare datetime.date to str

Reply:
Ok, I re-checked the _select_seller function and you are right. The expected date argument is string.

Reply:
@tfossoulw The problem is that in PO, date_order is Datetime, so the solution would be 

```python
date=po.date_order and fields.Date.to_string(fields.Date.from_string(po.date_order)) or None,
```